core/src/main/resources/org/apache/spark/ui/static/stagepage.js

```diff
@@ -404,8 +404,8 @@ $(document).ready(function () {
   
         var responseBody = response;
         var dataToShow = {};
-        dataToShow.showInputData = responseBody.inputBytes > 0;
-        dataToShow.showOutputData = responseBody.outputBytes > 0;
+        dataToShow.showInputData = responseBody.inputBytes > 0 || responseBody.inputRecords > 0 ;
+        dataToShow.showOutputData = responseBody.outputBytes > 0 || responseBody.outputRecords > 0;
```

Comment:
How to reproduce this issue?

Reply:
This can be easily reproduced by any spark structured streaming job reading from Kafka source.

Reply:
@wangyum Any more comments for this?

Reply:
Adding `responseBody.inputRecords > 0` seems correct. But why do we need to add `responseBody.outputRecords > 0`?

Reply:
if sending message into Kafka, output will have similar bug.